### PR TITLE
[MM-26868] Fix reselection of dropdowns on Interactive Dialogs

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
There was a regression on Interactive Dialogs where you were not able to change your selection on a drop down once you made the selection.

According to @ali-farooq0 #5845 was the offending commit, but on my local instance I was not able to get back the functionality removing the changes.

In order to solve this, I have added an update on the pretext whenever we clear the field.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-26868

#### Related Pull Requests
None

#### Screenshots
None

### CODE REVIEW

**Comment by larkox** [comment]
Note for QA: This touches the SuggestionBox component, so any suggestion box (channel switcher, autocomplete, searchbox...) might be affected by this change.

File: components/suggestion/suggestion_box.jsx [hunk 1]
```diff
@@ -517,6 +517,7 @@ export default class SuggestionBox extends React.PureComponent {
                 components: [],
                 selection: '',
             });
+            this.handlePretextChanged("");
```
Comment (ali-farooq0) [inline components/suggestion/suggestion_box.jsx:520]:
```suggestion
            this.handlePretextChanged('');
```

to fix the linting issue

**Comment by ali-farooq0** [comment]
Whoever tests this, make sure to test the following case:

#### Steps to reproduce:
1. Install [demo plugin](https://github.com/mattermost/mattermost-plugin-demo/releases/download/v07.0/com.mattermost.demo-plugin-0.7.0.tar.gz)
2. Run `/dialog` command
3. Click `Option Selector` 
4. Select an option
5. Attempt to select the same option again (it will not open)

2020-07-15 at 4 01 PM

**Comment by ali-farooq0** [comment]
Hey @prapti would you be able to help out with this?

**Comment by prapti** [comment]
In testing queue.

**Comment by ali-farooq0** [comment]
Excellent, thanks @prapti!

**Comment by amyblais** [comment]
@larkox Kind reminder to cherry-pick this,

**Pull request merged**